browser/brave_shields/ad_block_service_browsertest.cc

```diff
@@ -1789,6 +1792,38 @@ IN_PROC_BROWSER_TEST_F(CosmeticFilteringFlagDisabledTest,
   ASSERT_EQ(true, EvalJs(contents, "checkSelector('.ad', 'display', 'block')"));
 }
 
+// Ensure cosmetic filtering occurs always when AllowCosmeticFiltering() is
+// called.
+IN_PROC_BROWSER_TEST_F(AdBlockServiceTest, AllowCosmeticFiltering) {
+  ASSERT_TRUE(InstallDefaultAdBlockExtension());
+  GURL url =
+      embedded_test_server()->GetURL("b.com", "/cosmetic_filtering.html");
+
+  // Set cosmetic filtering is not enabled.
+  brave_shields::SetCosmeticFilteringControlType(
+      content_settings(), brave_shields::ControlType::ALLOW, url);
+  UpdateAdBlockInstanceWithRules("b.com###ad-banner");
+
+  // Used background web contents as AllowCosmeticFiltering() will be used for
```

Comment:
Could we use `PlaylistDownloadRequestManager::CreateWebContents` here? To avoid attaching webContentsObserver manually.